modules/claims_api/lib/bd/bd.rb

```diff
@@ -0,0 +1,81 @@
+# frozen_string_literal: true
+
+require 'evss_service/base'
+require 'faraday'
+
+module ClaimsApi
+  ##
+  # Class to interact with the BRD API
+  #
+  # Takes an optional request parameter
+  # @param [] rails request object
+  class BD
+    def initialize(request = nil)
+      @request = request
+    end
+
+    ##
+    # Upload document of mapped claim
+    #
+    # @return success or failure
+    def upload(claim, pdf_path, file_number)
+      body = generate_upload_body(claim, pdf_path, file_number)
+      client.post('documents', body)&.body
+    end
+
+    ##
+    # Generate form body to upload a document
+    #
+    # @return {paramenters, file}
+    def generate_upload_body(claim, pdf_path, file_number)
+      payload = {}
+      data = {
+        data: {
+          systemName: 'va.gov',
+          docType: 'L122',
+          claimId: claim.evss_id,
+          fileNumber: file_number,
+          fileName: File.basename(pdf_path),
+          trackedItemIds: []
+        }
+      }
+      payload[:parameters] = data
+      fn = Tempfile.new('params')
```

Comment:
`UploadIO` takes a `filename_or_io` - I feel like I'd ideally use an IO stream here instead, `Tempfile` will handle unique names, however, I don't think it'll remove the temp file until the pod restarts.

I guess I could merge `generate_upload_body` and `upload` so I can remove the tempfile myself after the `client.post`?